includes/utils/class-amp-validation-utils.php

```diff
@@ -272,4 +461,88 @@ public static function display_error( $response ) {
 		echo '</div>';
 	}
 
+	/**
+	 * Whether to validate the front end response.
+	 *
+	 * @return boolean
+	 */
+	public static function do_validate_front_end() {
+		$post = get_post();
+		return ( isset( $post ) && post_supports_amp( $post ) && self::has_cap() && ( isset( $_GET[ self::VALIDATION_QUERY_VAR ] ) ) ); // WPCS: CSRF ok.
+	}
+
+	/**
+	 * Adds the validation callback if front-end validation is needed.
+	 *
+	 * @param array $sanitizers The AMP sanitizers.
+	 * @return array $sanitizers The filtered AMP sanitizers.
+	 */
+	public static function add_validation_callback( $sanitizers ) {
+		if ( self::do_validate_front_end() ) {
+			foreach ( $sanitizers as $sanitizer => $args ) {
+				$args[ self::CALLBACK_KEY ] = __CLASS__ . '::track_removed';
+				$sanitizers[ $sanitizer ]   = $args;
+			}
+		}
+		return $sanitizers;
+	}
+
+	/**
+	 * Registers the post type to store the validation errors.
+	 *
+	 * @return void.
+	 */
+	public static function post_type() {
+		register_post_type(
+			self::POST_TYPE_SLUG,
+			array(
+				'labels'   => array(
+					'name' => _x( 'AMP Validation Errors', 'post type general name', 'amp' ),
+				),
+				'supports' => false,
```

Comment:
This should also have `'public' => false`, no?

Reply:
Thanks, [this commit](https://github.com/Automattic/amp-wp/commit/4955686933b6aaef1807cc1e1901b8a52c438415) adds  `'public' => false`.